cdap-api/src/main/java/io/cdap/cdap/api/annotation/Metadata.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotations can be used by entities to emit metadata.
+ * Currently the supported entity is plugin, the plugin with this annotation will have the metadata emitted when the
+ * plugin is deployed.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Metadata {
```

Comment:
In the cdap-api there is already a `Metadata` class, which carries tags and properties. So it seems like this annotation is a different way of emitting Metadata in addition to the use of `MetadataWriter`? Why the plugin cannot emit with the `MetadataWriter`?

Reply:
`MetadataWriter` can be used for plugins to emit metadata during program is running. Plugin with this annotation is able to emit metadata at the time it gets deployed. We simply need to inspect this annotation just like how we did for name, description, etc.